types/git-clone/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for git-clone 0.2
+// Project: https://github.com/jaz303/git-clone
+// Definitions by: Christoph Thiede <https://github.com/DefinitelyTyped>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="./promise" />
+
+/**
+ * Clone `repo` to `targetPath`, calling `cb` on completion.
+ */
+declare function clone(repo: string, targetPath: string, opts?: clone.Options, cb?: (error?: Error) => void): void;
+declare function clone(repo: string, opts?: clone.Options, cb?: (error?: Error) => void): void;
+
+declare namespace clone {
+    interface Options {
+        /**
+         * Path to git binary; defaults to `git`.
+         */
+        git?: string;
```

Comment:
For props like this one, please consider using `string | undefined`, if you want to 
add support for strict optional property types:
https://github.com/microsoft/TypeScript/pull/43947

Reply:
This is not clear to me, it passed the tests? Non-optional props cannot be omitted when calling the function which would change the semantics of the function.

Reply:
https://www.typescriptlang.org/play?ts=4.4.2&exactOptionalPropertyTypes=true&q=378#example/exact-optional-properties
it's about assigning undefined to an optional property, sometimes this should not be allowed, e.g. when underlying JS code would produce invalid results or event break.
This is optional, so skip this.